src/VisualStudio/Core/Impl/ProjectSystem/CPS/CPSProject_IWorkspaceProjectContext.cs

```diff
@@ -194,11 +195,13 @@ private AbstractProject GetAbstractProject(IWorkspaceProjectContext project)
         #endregion
 
         #region Files
-        public void AddSourceFile(string filePath, bool isInCurrentContext = true, IEnumerable<string> folderNames = null, SourceCodeKind sourceCodeKind = SourceCodeKind.Regular)
+        public void AddSourceFile(
+            string filePath, bool isInCurrentContext = true, IEnumerable<string> folderNames = null,
```

Comment:
So, is the expectation here that this will be a generated file on disk? This deviates a bit from what we discussed in the meeting, but I'm ok with trying this out to see what happens. 

What we had sketched out had us providing a `SourceTextContainer` instead. We can probably work with either of these approaches if necessary - I think the details matter more on the Roslyn side. 

---

It would require more work, but we could also turn this into something with a different behavior than `StandardTextDocument` if it makes sense.

Reply:
oh. I forgot that. I can do that as well. I can let you add in TextLoader

Reply:
actually, let's start with file generated. it doesn't need to because we can just let you add with SourceTextContainer or TextLoader or whatever you want to. but missing file can create some issue on some thing I dont know (since some components might expect file from VS  exist on disk)

so let's make it work with this first and then see we can just replace the call with SourceTextContainer or TextLoader and things still work. 

is that okay?